clients/src/main/java/org/apache/kafka/clients/producer/internals/BufferPool.java

```diff
@@ -228,7 +228,7 @@ public void deallocate(ByteBuffer buffer) {
     public long availableMemory() {
         lock.lock();
         try {
-            return this.availableMemory + this.free.size() * this.poolableSize;
+            return this.availableMemory + this.free.size() * (long) this.poolableSize;
```

Comment:
I think you missed that the second expression involves multiplication, not addition. So we need one of the values in the multiplication to be a long.